x-pack/apm-server/aggregation/txmetrics/aggregator_test.go

```diff
@@ -93,49 +93,59 @@ func TestTxnAggregatorProcessBatch(t *testing.T) {
 		// for 7 transactions and 3 services; first three service will receive
 		// 2 txns and the last one will receive 1 txn.
 		// Note that practically uniqueTxnCount will always be >= uniqueServices.
-		name                             string
-		uniqueTxnCount                   int
-		uniqueServices                   int
-		expectedActiveGroups             int64
-		expectedPerSvcTxnLimitOverflow   int
-		expectedOtherSvcTxnLimitOverflow int // we will design tests to overflow all the services equally
-		expectedTotalOverflow            int64
+		name                 string
+		uniqueTxnCount       int
+		uniqueServices       int
+		expectedActiveGroups int
+		// expectedOverflowReasonPerSvcTxnGrps represent total number of txn groups
+		// that overflowed due to per service txn group limit assuming all servies
+		// overflow equally. These will be recorded in the `transaction.name: _other`
+		// and the corresponding service name documents.
+		expectedOverflowReasonPerSvcTxnGrps int
+		// expectedOverflowReasonTxnGrps represent total number of txn groups that
+		// overflowed due to max txn groups limit. These will be recorded in the
+		// `transaction.name: _other` and the corresponding service name documents.
+		expectedOverflowReasonTxnGrps int
+		// expectedOverflowReasonSvc represents total number of txn groups that
+		// overflowed due to max services limit. These will be recorded in the
+		// `transaction.name: _other` and the `service.name: _other` document.
+		expectedOverflowReasonSvc int
 	}{
 		{
-			name:                             "record_into_other_txn_if_txn_per_svcs_limit_breached",
-			uniqueTxnCount:                   20,
-			uniqueServices:                   2,
-			expectedActiveGroups:             6,
-			expectedPerSvcTxnLimitOverflow:   8,
-			expectedOtherSvcTxnLimitOverflow: 0,
-			expectedTotalOverflow:            16,
+			name:                                "record_into_other_txn_if_txn_per_svcs_limit_breached",
+			uniqueTxnCount:                      20,
+			uniqueServices:                      2,
+			expectedActiveGroups:                6,
+			expectedOverflowReasonPerSvcTxnGrps: 16,
+			expectedOverflowReasonTxnGrps:       0,
+			expectedOverflowReasonSvc:           0,
 		},
 		{
-			name:                             "record_into_other_txn_if_txn_grps_limit_breached",
-			uniqueTxnCount:                   60,
-			uniqueServices:                   20,
-			expectedActiveGroups:             40,
-			expectedPerSvcTxnLimitOverflow:   2,
```

Comment:
why is this 0 now? (same for the above example)

Reply:
I changed the meaning behind the variables. Previously the value `expectedPerSvcTxnLimitOverflow` represented the total number of events that were recorded in the overflow bucket. These included the overflow due to max transaction limit as well as due to per service transaction limit (since if the max transaction limit is breached we will still record the value in the overflow bucket of the specific service). In this PR, I updated the variables to represent the overflow due to a specific limit breach. The older and newer values can be translated as:

`expectedPerSvcTxnLimitOverflow = expectedOverflowReasonPerSvcTxnGrps + expectedOverflowReasonTxnGrps`
`expectedOtherSvcTxnLimitOverflow = expectedOverflowReasonSvc`